### PR TITLE
fix(vscode): match correct colors

### PR DESCRIPTION
**Pull request by zam157**
Fixes:
- matching wrong colors when using 'dark:' prefix
- color preview not working on valueless attributes when `AttributifyOptions.strict = true`

### CODE REVIEW

**Comment by antfu** [comment]
Can you provide more examples of what cases you are fixing?

**Comment by zam157** [comment]
Sorry for the lack of descriptions. 

In current version of uno vscode extension, if you use `dark:` prefix with custom theme colors in attributify mode, 
```
// unocss.config.ts
  theme: {
    colors: {
      veryCool: '#0000ff',
      notBad: '#ffa321',
      brand: {
        primary: 'hsla(var(--hue, 217), 78%, 51%, 0.4)',
        success: 'rgba(255, 234, 123, 0.4)',
      },
    },
  },
```
it would always match `dark` color:
image
Fix:
image

**Comment by zam157** [comment]
Another case:

Color preview not working on valueless attributes when `AttributifyOptions.strict = true`
```
// unocss.config.ts
  presets: [
    presetUno(),
    presetAttributify({
      strict: true,
    }),
  ],
```
image

Fix:
image

File: packages/vscode/src/utils.ts [hunk 1]
```diff
@@ -56,27 +57,34 @@ export function body2ColorValue(body: string, theme: Theme) {
...
+    if (cachedColor) {
+      colorsMap.set(colorKey, cachedColor)
       continue
     }
 
-    const matchedAttr = i.match(matchedAttributifyRE)
-    const body = matchedAttr?.[0] ?? i // remove prefix e.g. `dark:` `hover:`
+    const matchedAttr = i.match(matchedAttributifyRE)?.[0] ?? matchedValueless
+    const body = (matchedAttr ?? i)
+      .split(':').at(-1) ?? '' // remove prefix e.g. `dark:` `hover:`
```
Comment (antfu) [inline packages/vscode/src/utils.ts:79]: `.at` syntax is too new. Let's use `.slice(-1)[0]` instead.

**Pull request merged**